RyanLeys

heyryanleys opened the pull request:
My infographic

heyryanleys commented:
@FranDias 
![screen shot 2016-02-28 at 5 28 37 pm](https://cloud.githubusercontent.com/assets/16675632/13382360/c169543e-de40-11e5-9ea7-f973a9392feb.png)

FranDias commented on README.md:
```diff
@@ -1,6 +1,9 @@
-- show off infographics
+- I couldn't figure out how to shift the graph over so I had to use filler blank data as a work around
+
+- I'm not sure what the best way to show "0" on the graph would be, but I think it's necessary to have
```
What about a horizontal line in a different color?

FranDias commented on README.md:
```diff
@@ -1,6 +1,9 @@
+# Ryan's Infographic
 
-- show off infographics
+- I couldn't figure out how to shift the graph over so I had to use filler blank data as a work around
```
You can add a constant to the `x` to push it over.

FranDias commented on README.md:
```diff
@@ -1,6 +1,9 @@
+
+- I'm not sure what the best way to show "0" on the graph would be, but I think it's necessary to have
+
+- I kind of want to make it so the color changes as the graph goes along, but I didn't know how to do that
```
You probably want to use a predefined set of colors at first. It'll just be checking a range & changing a variable.

FranDias commented on README.md:
```diff
@@ -1,6 +1,9 @@
+
+- I kind of want to make it so the color changes as the graph goes along, but I didn't know how to do that
+
+- I possibly want to make it so when you hover over it you have the date show up. I might just add the date underneath them too. 
```
We're going to make the graphs interactive in a few classes (using constructor functions)

FranDias commented:
Nice work so far!

heyryanleys commented:
![screen shot 2016-03-01 at 4 00 36 pm](https://cloud.githubusercontent.com/assets/16675632/13441780/0a093a92-dfc7-11e5-9854-110943db673d.png)

@FranDias 

- Refactored Code
- Added Stars
- Added Moon
- Added line to show where 0 is

FranDias commented on sketch.js:
```diff
@@ -0,0 +1,206 @@
+
+// Adding Line at 0
+  stroke(300);
+  line(200, 250, 800, 250);
```
Maybe tuck this guy under your content or add an opacity? It's useful but seems kind of overpowering.

heyryanleys commented:
@FranDias 

- Added interactivity to the infographic (when you hover it shows date and interest level)
- Moved placement on the line based on your feedback
- Further simplified some code

![screen shot 2016-03-22 at 12 35 47 pm](https://cloud.githubusercontent.com/assets/16675632/13959577/0d9db27c-f02b-11e5-94b7-911709fdbd7b.png)

However I'm having a really hard time getting it so after you hover off the bar it disappears. I've tried if/else and different functions, but I can't seem to get it right. It ends up displaying all the dates like this:

![screen shot 2016-03-22 at 12 36 17 pm](https://cloud.githubusercontent.com/assets/16675632/13959604/29ba0d7a-f02b-11e5-9bf8-2048a1e8c66e.png)

FranDias commented on sketch.js:
```diff
@@ -0,0 +1,92 @@
+
+var graph = function (someData) {
+  for (var i = 0; i < someData.length; i++){
+    var barWidth = 13
```
This doesn't change inside the loop so it can be declared outside of it.

FranDias commented on sketch.js:
```diff
@@ -0,0 +1,92 @@
+      ellipse(this.x, this.y, 5, 5);
+      }
+      this.move = function(){
+        this.x = this.x + random (0, 0);
```
`random` Doesn't actually do anything here. Also, try to stay consistent formatting (take away the space between `random` and `(0,0)`

FranDias commented on sketch.js:
```diff
@@ -0,0 +1,92 @@
+      }
+    }
+
+// Adding Moon
```
You should put this into a helper function `renderMoon` or something similar, just group it together.

FranDias commented on sketch.js:
```diff
@@ -0,0 +1,92 @@
+      var spd = myData[i];
+      var barWidth = 13
+      var spacing = i * (5 + barWidth);
+    if (abs((spacing + (barWidth / 2)) - mouseX) < (barWidth / 2) && spd.date != "filler") {
```
Use collide for this or abstract this out into it's own function.

FranDias commented on sketch.js:
```diff
@@ -0,0 +1,92 @@
+      rect(0, 175, width, 60)
+      noStroke()
+      fill(34,148,143) // Blue for Histogram
+      graph(myData);
```
Instead of just graphing the data you should be instantiating an instance for each point like you did with `Stars`

FranDias commented:
Overall nice work. Couple minor things but good all around.
